cosmo_tester/test_suites/image_based_tests/simple_agent_upgrade_test.py

```diff
@@ -15,56 +15,65 @@
 
 import pytest
 
-from cosmo_tester.framework.test_hosts import TestHosts
-from cosmo_tester.framework.examples.nodecellar import NodeCellarExample
+from cosmo_tester.framework import util
+from cosmo_tester.framework.examples import get_example_deployment
+from cosmo_tester.framework.test_hosts import TestHosts as Hosts
 
 from cosmo_tester.test_suites.snapshots import restore_snapshot
 
+ATTRIBUTES = util.get_attributes()
+
 
 @pytest.fixture(scope='module')
-def managers(cfy, ssh_key, module_tmpdir, attributes, logger):
-    hosts = TestHosts(cfy, ssh_key, module_tmpdir, attributes, logger, 2)
+def managers_and_vm(cfy, ssh_key, module_tmpdir, attributes, logger):
+    hosts = Hosts(cfy, ssh_key, module_tmpdir, attributes, logger, 3)
     try:
-        _managers = hosts.instances
+        managers = hosts.instances[:2]
+        vm = hosts.instances[2]
+
+        managers[0].upload_files = False
+        managers[1].upload_files = False
+        managers[0].restservice_expected = True
+        managers[1].restservice_expected = True
+
+        vm.upload_files = False
+        vm._image_name = ATTRIBUTES['centos_7_image_name']
+        vm._linux_username = ATTRIBUTES['centos_7_username']
 
-        # The second manager needs to be clean, to allow restoring to it
-        _managers[1].upload_plugins = False
         hosts.create()
-        hosts.instances[0].upload_plugin(
-            attributes['default_openstack_plugin']
-        )
-        yield _managers
+        yield hosts.instances
     finally:
         hosts.destroy()
 
 
 @pytest.fixture(scope='function')
-def nodecellar(managers, cfy, ssh_key, tmpdir, attributes, logger):
-    """
-    Using nodecellar instead of hello world, because the process stays up
-    after the old agent is stopped (as opposed to the webserver started in
-    hello world)
-    """
-    manager = managers[0]
-    nc = NodeCellarExample(cfy, manager, attributes, ssh_key, logger, tmpdir)
-    nc.blueprint_file = 'openstack-blueprint.yaml'
-    yield nc
-    if nc.cleanup_required:
-        nc.cleanup()
+def example(managers_and_vm, cfy, ssh_key, tmpdir, attributes, logger):
+    manager = managers_and_vm[0]
+    vm = managers_and_vm[2]
+
+    example = get_example_deployment(
+        cfy, manager, ssh_key, logger, 'agent_upgrade', vm)
+
+    try:
+        yield example
+    finally:
+        if example.installed:
+            example.uninstall()
 
 
 def test_old_agent_stopped_after_agent_upgrade(
-        managers, nodecellar, cfy, logger, tmpdir
+        managers_and_vm, example, cfy, logger, tmpdir
 ):
     local_snapshot_path = str(tmpdir / 'snapshot.zip')
     snapshot_id = 'snap'
 
-    old_manager = managers[0]
-    new_manager = managers[1]
+    old_manager = managers_and_vm[0]
```

Comment:
you can btw do `old_manager, new_manager, vm = managers_and_vm` and that of course also implicitly checks that managers_and_vm is indeed 3 entries

Reply:
Fixing